### PR TITLE
sycl: initial cmake support of SYCL for AMD GPUs

### PR DESCRIPTION
**Pull request by Alcpz**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

---

The SYCL backend can be run in AMD GPUs as well as long as the `sub_group_size` (i.e. WARP_SIZE) is either 16 or 32. 
This PR gives initial support to compile on AMD GPUs. Since the backend does not provide full support, I don't think it's good to update the documentation until it is improved.

### CODE REVIEW

**Review by OuadiElfarouki — APPROVED** [review]
LGTM. I would only suggest to add a comment/description on the gfx/arch or devices tested. Thank you!

File: ggml/src/CMakeLists.txt [hunk 1]
```diff
@@ -543,6 +546,12 @@ if (GGML_SYCL)
 
     if (GGML_SYCL_TARGET STREQUAL "NVIDIA")
         add_compile_definitions(GGML_SYCL_WARP_SIZE=32)
+    elseif (GGML_SYCL_TARGET STREQUAL "AMD")
+        # INFO: Allowed Sub_group_siezs are not consistent through all
```
Comment (joeatodd) [inline ggml/src/CMakeLists.txt:550]:
```suggestion
        # INFO: Allowed Sub_group_sizes are not consistent through all
```

**Comment by joeatodd** [comment]
Looks good. Could we describe the AMD support in the SYCL README?

**Review by joeatodd — APPROVED** [review]

**Comment by Alcpz** [comment]
> Looks good. Could we describe the AMD support in the SYCL README?

@joeatodd I´ve added the documentation for the backend.

File: docs/backend/SYCL.md [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 ### Llama.cpp + SYCL
 
-The llama.cpp SYCL backend is designed to support **Intel GPU** firstly. Based on the cross-platform feature of SYCL, it could support other vendor GPUs: Nvidia GPU (*AMD GPU coming*).
+The llama.cpp SYCL backend is designed to support **Intel GPU** firstly. Based on the cross-platform feature of SYCL, it could support other vendor GPUs: Nvidia and AMD GPU.
```
Comment (joeatodd) [inline docs/backend/SYCL.md:29]:
```suggestion
The llama.cpp SYCL backend is designed to support **Intel GPU** firstly. Based on the cross-platform feature of SYCL, it also supports other vendor GPUs: Nvidia and AMD GPU.
```

**Review by joeatodd — APPROVED** [review]
Thanks for the docs!

**Pull request merged**